cql3/type_json.cc

```diff
@@ -561,3 +561,48 @@ sstring to_json_string(const abstract_type& t, bytes_view bv) {
 sstring to_json_string(const abstract_type& t, const managed_bytes_view& mbv) {
     return visit(t, to_json_string_visitor{linearized(mbv)});
 }
+
+namespace {
+struct to_json_type_visitor {
+    managed_bytes_view bv;
+    rjson::type operator()(const reversed_type_impl& t) { return to_json_type(*t.underlying_type(), bv); }
+    template <typename T> rjson::type operator()(const integer_type_impl<T>& t) { return rjson::type::kNumberType; }
+    template <typename T> rjson::type operator()(const floating_type_impl<T>& t) {
+        if (bv.empty()) {
+            throw exceptions::invalid_request_exception("Cannot create JSON string - deserialization error");
+        }
+        auto v = t.deserialize(linearized(bv));
+        T d = value_cast<T>(v);
+        if (std::isnan(d) || std::isinf(d)) {
+            return rjson::type::kNullType;
+        }
+        return rjson::type::kNumberType;
+    }
+    rjson::type operator()(const uuid_type_impl& t) { return rjson::type::kStringType; }
+    rjson::type operator()(const inet_addr_type_impl& t) { return rjson::type::kStringType; }
+    rjson::type operator()(const string_type_impl& t) { return rjson::type::kStringType; }
+    rjson::type operator()(const bytes_type_impl& t) { return rjson::type::kStringType; }
+    rjson::type operator()(const boolean_type_impl& t) {
+        const auto val = t.to_string(linearized(bv));
+        return val == "true" ? rjson::type::kTrueType : rjson::type::kFalseType;
```

Comment:
I was shocked that RapidJSON has a different type for false and true. Good they don't have a different type for the number 3 and the number 7 :-)

Reply:
By the way, I'm betting there is a more efficient way to check if t contains a true and a false than converting it to a string and checking the string against the strings "true" and "false", but I guess this performance of this isn't terribly important anyway.

Reply:
> I was shocked that RapidJSON has a different type for false and true. Good they don't have a different type for the number 3 and the number 7 :-)

Look up std::true_type and std::false_type.